### PR TITLE
chore: standardize repository for procore open-source

### PR DESCRIPTION
**Pull request by jmeridth**
- [x] setup .github folder (actions, dependabot.yaml, ISSUE_TEMPLATES, PULL_REQUEST_TEMPLATE.md, CODEOWNERS)
  - [x] add procore-oss/procore-rspec-profiling team to CODEOWNERS
- [x] setup required files (README.md, SECURITY.md, CODE_OF_CONDUCT.md, LICENSE.md, CONTRIBUTING.md)
- [x] update gemspec file
- [x] setup postgres in tests in GitHub actions
  - [x] test against postgres bookworm versions
- [x] add EMERITUS mentioning former upstream contributors
- [x] update LICENSE.md with previous owners
- [x] setup dummy rails app in specs folder

Next PRs after this:

- up default ruby version
- add Makefile and docker-compose for local dev to run rspec
- possible better solution other than dummy rails app

Checklist:

* [x] I have updated the necessary documentation
* [x] I have signed off all my commits as required by [DCO](https://GitHub.com/apps/dco/)
* [x] My build is green

### CODE REVIEW

File: .github/CODEOWNERS
Comment (pc-bob) [inline .github/CODEOWNERS]: do we want to add the [procore-rspec-profiling](https://github.com/orgs/procore-oss/teams/procore-rspec-profiling) team as codeowners here?

File: .github/workflows/release.yaml
Comment (pc-bob) [inline .github/workflows/release.yaml]: blueprinter?

File: .github/workflows/test.yaml
Comment (pc-bob) [inline .github/workflows/test.yaml]: should we remove usage of the term `master`?

File: SECURITY.md
Comment (pc-bob) [inline SECURITY.md:13]: we are also validating 3.3 in `test.yaml`

File: SECURITY.md
Comment (pc-bob) [inline SECURITY.md]: blueprinter

File: rspec_profiling.gemspec
Comment (pc-bob) [inline rspec_profiling.gemspec]: Prodocore

File: spec/dummy/README.md
Comment (pc-bob) [inline spec/dummy/README.md]:
```suggestion
This is a dummy Rails application we use to test the gem. It is not meant to be used as a real application.
```

File: .github/CODEOWNERS
Comment (jmeridth) [reply .github/CODEOWNERS]: I hadn't created the team yet because who will be on it?  Do we know who the owners are yet?  I can create it and add myself and you for now?

File: .github/workflows/release.yaml
Comment (jmeridth) [reply .github/workflows/release.yaml]: Good catch.  Dang copy/paste 😄

File: .github/workflows/test.yaml
Comment (jmeridth) [reply .github/workflows/test.yaml]: yes. doing.

File: .github/CODEOWNERS
Comment (jmeridth) [reply .github/CODEOWNERS]: I take it back.  Adding.

**Comment by pc-bob** [comment]
should we update the `ruby-version` file to update the default version of ruby used for the project now that we're testing against more modern versions?

**Comment by jmeridth** [comment]
@pc-bob yes, after this merges IMO.  Updating description with further PR changes after this merges.

**Review by pc-bob — APPROVED** [review]

**Pull request merged**
